gc/base/MemoryPoolAddressOrderedList.hpp

```diff
@@ -39,7 +39,13 @@ class MM_AllocateDescription;
 class MM_ConcurrentSweepScheme;
 #endif /* OMR_GC_CONCURRENT_SWEEP */
 
+#if defined(J9ZOS390) && !defined(OMR_ENV_DATA64)
+/* Z31 */
+#define FREE_ENTRY_END ((MM_HeapLinkedFreeHeader *) I_32_MAX)
+#else
 #define FREE_ENTRY_END ((MM_HeapLinkedFreeHeader *) UDATA_MAX)
+#endif
+
```

Comment:
but there is already a similar define that (OMRPORT_VMEM_MAX_ADDRESS) we could already use are (for being more concise pre-define it) @dmitripivkine ?

#define FREE_ENTRY_END OMRPORT_VMEM_MAX_ADDRESS

Reply:
we have a specific type (MM_HeapLinkedFreeHeader), so we have to define our own

#define FREE_ENTRY_END ((MM_HeapLinkedFreeHeader *)OMRPORT_VMEM_MAX_ADDRESS)